### PR TITLE
Correct Sidekiq configuration

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* Sidekiq jobs were not working correctly in development for a few reasons:
1. The interface for `IndexerJob` was not being used correctly.
1. There was no `worker` line in the `Procfiles`, so no workers were initiated. Hence, no jobs were being done.
1. Sidekiq was not initialized to use the correct redis connection URL.

Changes
=======
* Add configuration files and initializer for sidekiq for correct functioning.
* Add worker directives to foreman's `Procfile` manifest.
* Correct an interface error when creating `IndexerJob`s.

Minor
-----
* The "hey you've disabled the Elasticsearch callbacks" warning was inverted and incorrect. Fixed here.
* Add the sidekiq web monitoring page at `/sidekiq`, but only in development.

https://trello.com/c/f0lpURT0/27-add-a-search-field-and-return-a-result

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1 +1,2 @@
-web: bundle exec puma -C config/puma.rb
\ No newline at end of file
+web: bundle exec puma -C config/puma.rb
+worker: bundle exec sidekiq -C config/sidekiq.yml
```
Comment (ptrikutam) [inline Procfile]: We'll probably need to start paying for a worker on Heroku :(

File: app/service/search_index.rb [hunk 2]
```diff
@@ -1,23 +1,27 @@
 class SearchIndex
   def self.add(record)
     if search_index_callbacks_enabled?
-      IndexerJob.perform_async(:index, record.id)
+      IndexerJob.perform_async(:index, record.class.name, record.id)
```
Comment (ptrikutam) [inline app/service/search_index.rb]: Yeesh, can't believe I missed this. Good catch.

File: Procfile.dev [hunk 3]
```diff
@@ -1,2 +1,3 @@
 web: bundle exec puma -C config/puma.rb
+worker: bundle exec sidekiq -C config/sidekiq.yml
```
Comment (ptrikutam) [inline Procfile.dev]: Can we also add a process for redis?

```
redis: redis-server
```

Comment (ptrikutam) [inline Procfile.dev]: We may also want to add `dump.rdb` to the `.gitignore` when we do this.

**Comment by adarsh** [comment]
> EDIT: I checked out a different branch and restarted foreman, and jobs started firing. I then ran into this issue (after previously manually deleting record 100): https://cl.ly/2x2c1r1I0W26

So that pic is showing errors where the job is failing because it's trying to find record 100 but it doesn't exist. That's a bug for us/me - I'm asking ES to remove a record from the index which does not exist in the database anymore :(

I'm opening an issue for this. Solution is something like `ResourceDeleter.remove` which will first remove the record from the index, *then* delete the record.

Good catch! Made an issue here: https://github.com/greencommons/commons/issues/43

File: Procfile.dev (hunk 3 above)
Comment (adarsh) [inline Procfile.dev]: Yeah good call. Done in dd48e3b

**Comment by adarsh** [comment]
@ptrikutam note that the other branch is a PR into this branch, so I will **merge it second**.

**Pull request merged**